subprojects/build-cache-base/src/main/java/org/gradle/caching/internal/origin/OriginMetadata.java

```diff
@@ -16,15 +16,13 @@
 
 package org.gradle.caching.internal.origin;
 
-import com.google.common.base.Preconditions;
-
 public class OriginMetadata {
 
     private final String buildInvocationId;
     private final long executionTime;
 
     public OriginMetadata(String buildInvocationId, long executionTime) {
-        this.buildInvocationId = Preconditions.checkNotNull(buildInvocationId, "buildInvocationId");
+        this.buildInvocationId = buildInvocationId;
```

Comment:
⭕️ You can use `Objects.requireNonNull()` if you want to keep this assertion (which I think is a good thing for data objects)

Reply:
I know, but we don't actually check anywhere on internal APIs for `null` parameters. I'm also not a big fan of defensive code like this, especially since we have nullability checks now in IntelliJ.